app/controllers/application_controller.rb

```diff
@@ -1,18 +1,27 @@
 class ApplicationController < ActionController::Base
-  before_action :set_current_user
+  before_action :authenticate_user!
+  protect_from_forgery with: :exception
+  helper_method :current_user
+
+  before_action :update_allowed_parameters, if: :devise_controller?
 
   private
 
-  def set_current_user
-    user_id = params[:user_id]
-    @current_user = User.find_by(id: user_id)
+  def current_user
+    @current_user ||= warden.user
   end
```

Comment:
- [ ] Great job working on this milestone. As per project requirements, the current_user method should be removed as devise provides a way of doing this.
![Screen Shot 2023-09-07 at 10 20 34 AM](https://github.com/Nahid1911/blog-app-ruby-on-rails/assets/62849872/c63cf8dc-b5f5-4031-a1e0-6ef9703b28ec)

Reply:
resolved